src/frontends/onnx/frontend/src/op/com.microsoft/matmulnbits.cpp

```diff
@@ -0,0 +1,228 @@
+    const auto bits = node.get_attribute_value<int64_t>(
+        "bits",
+        4);  // required, in docs: number of bits used for weight quantization (default 4)
+
+    const uint64_t n_blocks_per_col = (K + block_size - 1) / block_size;
+    const auto blob_size = static_cast<int64_t>(ceil(block_size * bits / 8));
+
+    CHECK_VALID_NODE(node, n_blocks_per_col > 0, "Wrong blocks count: ", n_blocks_per_col);
+    CHECK_VALID_NODE(node, blob_size > 0, "Wrong blob size: ", blob_size);
+    // in documentation: ...Input B is a 2D constant Matrix.
+    CHECK_VALID_NODE(node,
+                     dynamic_cast<v0::Constant*>(b_quantized.get_node()) != nullptr,
+                     "MatMulNBits limitation: accepting only a constant as a B input");
+    CHECK_VALID_NODE(node,
+                     b_quantized.get_partial_shape().rank() == 3,
+                     "Expected rank of quantized weights is 3 [N][n_blocks_per_col][blob_size], got: ",
+                     b_quantized.get_partial_shape().rank());
+    CHECK_VALID_NODE(node,
+                     a.get_element_type() == ov::element::f16 || a.get_element_type() == ov::element::f32,
+                     "Unsupported input A type, accepted FP16, FP32, got: ",
+                     a.get_element_type());
+    CHECK_VALID_NODE(
+        node,
+        b_quantized.get_element_type() == ov::element::u8 || b_quantized.get_element_type() == ov::element::i32,
+        "Unsupported input B type, accepted FP16, FP32, got: ",
+        b_quantized.get_element_type());
+
+    CHECK_VALID_NODE(node,
+                     block_size >= 16 && (block_size % 2 == 0),
+                     "Wrong block size, should be >=16 and be a power of 2, got: ",
+                     block_size);
+    CHECK_VALID_NODE(node, accuracy_level >= 0 && accuracy_level <= 4, "Unsupported accuracy level: ", accuracy_level);
+
+    if (inputs.size() > 3) {
+        zero_points = inputs[3];
+        CHECK_VALID_NODE(node,
+                         zero_points.get_element_type() == ov::element::u8 ||
+                             zero_points.get_element_type() == ov::element::i32 ||
+                             zero_points.get_element_type() == ov::element::f32 ||
+                             zero_points.get_element_type() == ov::element::f16,
+                         "Unsupported input zero_points type, accepted U8, I32, FP16, FP32, got: ",
+                         zero_points.get_element_type());
+    }
+
+    if (inputs.size() > 4) {
+        group_idx = inputs[4];
+        CHECK_VALID_NODE(node,
+                         group_idx.get_element_type() == ov::element::i32,
+                         "Unsupported input group_idx type, accepted I32, got: ",
+                         group_idx.get_element_type());
+    }
+
+    if (inputs.size() > 5) {
+        bias = inputs[5];
+        CHECK_VALID_NODE(node,
+                         bias.get_element_type() == a.get_element_type(),
```

Comment:
Should we really check it? Can we assume that they are the same by specification in correct model?

Reply:
also, not good because `get_element_type()` can return dynamic or undefined type for some tensors.

Reply:
Docs says they should have same types. Yes, I know about possible issue with dynamic type. But currently it fails at a constant creation step - I cannot create "dynamic" constant for the default_zp. I'm trying remember, did you do something similar like "ConstantLike" in TF?